### DISCUSSION TITLE
Moving Files in External Library

### DISCUSSION THREAD

**Discussion by itzTheMeow** [immich-app/immich#10342] [Q&A]
I have an external library set to a folder that i keep my images in.
I don't upload to immich directly, i keep my images in a folder on my NAS.
When i move images from my phone into the folder, immich usually scans and adds them to the library.
However, i have a script that runs to organize the files into year/month instead of having them all in one folder.
When the script runs, immich re-scans and now there's duplicate images because their locations changed. The old images are marked as offline and i have to remove offline files on the library.

I was just doing that, however with the new duplicate detection, it's detecting the offline files as duplicates and when i remove them, the entries don't get removed from the duplicate detection (so it's just a choice between 1 file).
So now it's a more annoying issue.

Wondering if there's any fix for this?

**Comment by Elerir** [comment]
Hello,  
I'm facing the same issue :  
I upload my pictures through Nextcloud, and I connected my photo repository as an external library  
If i move a picture through Nextcloud, I'll have a duplicate on immich, and the previous location will be set at "not found"  
"Scanning library" doesn't fix the issue.
I think there should be a "scan all library" checking for moved files

**Comment by Daniel-Kalus** [comment]
[Here](https://immich.app/docs/features/libraries/) it says moving files inside an external library is not possible right now without issues like mentioned above or losing metadata, but will be in a future release. I'm waiting eagerly for this future release, because it would make external libraries perfect. It would mean I could organize my pictures in folders the way I like it, reorganizing when new pictures are added or when I change my mind, and still keep metadata information like detected faces in Immich. How wonderful 🙏

**Reply by eharris** [reply]
Is there a ticket for that feature/improvement?  If so I can't find it.

**Reply by ssynnes** [reply]
Here is the discussion on the topic: https://github.com/immich-app/immich/discussions/16394

**Comment by mburyakov** [comment]
Is there any ticket?

I'd extremely vote for it because impossibility to move/rename files within external library sticks me to my current (bad) file layout. 

Why may I want to change file layout in future:
- If file or directory names contain platform-dependent (like `:` or `"`) or non-ascii characters (like cyrillic `й` which is encoded differently on MacOs). When migrating to other os, files have to be renamed.
- If external library has its own logic of file structuring, but a part of it is structured differently or incorrectly due to oversight.
- If I have multiple external libraries and want to move files between them.
- If I want to move files from uploaded library to external library or vice versa (it's another useful feature but not so necessary for me personally)

Also, detecting file moves feature is rather similar to detecting duplicates. Even more, duplicate detection can be a workaround for move detection: if new asset will be added to the same albums etc as the original one, it's almost enough (but then probably the trash have to be emptied manually with care).
